e2e/win7/ff32 date range test fails

pData\Local\Temp\sc-launcher-readyfile
02 Feb 14:50:38 - Using no proxy for connecting to Sauce Labs REST API.
02 Feb 14:50:38 - Resolving saucelabs.com to 162.222.73.28 took 93 ms.
02 Feb 14:50:38 - ***********************************************************
02 Feb 14:50:38 - A newer version of Sauce Connect (build 1630) is available!
02 Feb 14:50:38 - Download it here:
02 Feb 14:50:38 - https://saucelabs.com/downloads/sc-4.3.6-win32.zip
02 Feb 14:50:38 - ***********************************************************
02 Feb 14:50:39 - Started scproxy on port 52687.
02 Feb 14:50:39 - Please wait for 'you may start your tests' to start your tests.
02 Feb 14:50:39 - Starting secure remote tunnel VM...
[14:50:40] saw 217 files
[14:50:40] Finished 'build' after 3.85 s
02 Feb 14:50:40 - Checking
domain overlap for my domain sauce-connect.proxy, other tunnel domain sauce-connect.proxy
02 Feb 14:50:40 - Overlapping domain: sauce-connect.proxy, shutting d
own tunnel efc24bcde2034ccab6f0f86d962bd722.
02 Feb 14:50:48 -
Secure remote tunnel VM provisioned.
02 Feb 14:50:4
8 - Tunnel ID: e5dc262a62e54988b77ad940da1c8d81
02 Feb 14:50:48 - Secure remote tunnel VM is now: b
ooting
02 Feb 14:50:51 - Secure remo
te tunnel VM is now: running
02 Feb 14:50:51 - Remote tunnel host is: maki77060.miso.saucelabs.com
02 Feb 14:50:51 - Using no proxy for co
nnecting to tunnel VM.
02 Feb 14:50:51 - Re
solving maki77060.miso.saucelabs.com to 162.222.77.60 took 0 ms.
02 Feb 14:50:51 - Starting S
elenium listener...
02 Feb 14:50:51 - Establishing secure TLS connection to tunnel...
02 Feb 14:50:51 -
Selenium listener started on port 4445.
02 Feb 14:50:54 - Sauce Connect
is up, you may start your tests.
02 Feb 14:50:54 - Connection established.

[14:50:58] Sauce Connect ready
[14:50:58] Finished 'selenium-start' after 22 s
WARNING - You have specified both capabilites and multiCapabilities. This will result in capabilities being ignored
Using SauceLabs selenium server at http://localhost:4445/wd/hub
[launcher] Running 1 instances of WebDriver
@landing-page
Feature: Landing Page

  Users who visit the "home page" of the application should see the explore
  state.


  Scenario: The page title is correct              # :\Users\builder\marklogic-samplestack\specs\features\as-anyone\landing-page.feature:7
    When I visit the "landing" page                # :\Users\builder\marklogic-samplestack\specs\features\as-anyone\landing-page.feature:8
    Then the page title is "explore - samplestack" # :\Users\builder\marklogic-samplestack\specs\features\as-anyone\landing-page.feature:9

@explore-docs-by-resolved-only @explore
Feature: Explore Docs By Resolved Only

  When searching for documents, a facet filter may be applied to limit results to
  those documents which are only resolved.


  Scenario: As a contributor filtering by resolved only, I see the correct results              # :\Users\builder\marklogic-samplestack\specs\features\as-contributor\explore-docs-by-resolved-only.feat
ure:7
    Given I am a contributor                                                                    # :\Users\builder\marklogic-samplestack\specs\features\as-contributor\explore-docs-by-resolved-only.feat
ure:8
    And I am using the brief seed data                                                          # :\Users\builder\marklogic-samplestack\specs\features\as-contributor\explore-docs-by-resolved-only.feat
ure:9
    And I visit the "explore" page                                                              # :\Users\builder\marklogic-samplestack\specs\features\as-contributor\explore-docs-by-resolved-only.feat
ure:10
    And I clear all filters                                                                     # :\Users\builder\marklogic-samplestack\specs\features\as-contributor\explore-docs-by-resolved-only.feat
ure:11
    When I filter documents by resolved only = "true"                                           # :\Users\builder\marklogic-samplestack\specs\features\as-contributor\explore-docs-by-resolved-only.feat
ure:12
    Then the docs count is "1903"                                                               # :\Users\builder\marklogic-samplestack\specs\features\as-contributor\explore-docs-by-resolved-only.feat
ure:13
    When I focus on the "first" search result                                                   # :\Users\builder\marklogic-samplestack\specs\features\as-contributor\explore-docs-by-resolved-only.feat
ure:14
    Then the result "title" is "Q: Mary's Question Number 0"                                    # :\Users\builder\marklogic-samplestack\specs\features\as-contributor\explore-docs-by-resolved-only.feat
ure:15
    When I focus on the "last" search result                                                    # :\Users\builder\marklogic-samplestack\specs\features\as-contributor\explore-docs-by-resolved-only.feat
ure:16
    Then the result "title" is "Q: Show modal if page refreshed, but not if a link is followed" # :\Users\builder\marklogic-samplestack\specs\features\as-contributor\explore-docs-by-resolved-only.feat
ure:17

@explore-docs-by-date-range @explore
Feature: Explore Docs By Date Range

  When searching for documents, a facet filter may be applied to limit results to
  those documents which created by the user.


  Scenario: As a contributor filtering by date range, I see the correct results                                 # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:7
    Given I am "Joe"                                                                                            # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:8
    When I visit the "explore" page                                                                             # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:9
    And I clear all filters                                                                                     # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:10
    When I filter documents by to date = "12/01/2012"                                                           # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:11
    And I press key enter in to date                                                                            # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:12
    Then the docs count is "1181"                                                                               # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:13
      AssertionError: expected 1903 to equal 1181


    When I focus on the "first" search result,                                                                  # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:14
    Then the result "title" is "Q: JS split() function that ignores separator appearing inside quotation marks" # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:15
    When I filter documents by from date = "01/01/2011"                                                         # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:16
    And I press key enter in from date                                                                          # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:17
    Then the docs count is "1004"                                                                               # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:18
    When I focus on the "last" search result,                                                                   # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:19
    Then the result "title" is "Q: Javascript: Easiest way to make a list of elements"                          # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:20

@explore-docs-by-mine-only @explore
Feature: Explore Docs By Mine Only

  When searching for documents, a facet filter may be applied to limit results to
  those documents which created by the user.


  Scenario: As a contributor filtering by mine only, I see the correct results   # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-mine-only.feature:7
    Given I am "Joe"                                                             # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-mine-only.feature:8
    When I visit the "ask" page                                                  # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-mine-only.feature:9
    And I type "mine only test" as the question title                            # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-mine-only.feature:10
    And I type "**test**" as the question content                                # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-mine-only.feature:11
    And I enter "e2eTests" as a question tag                                     # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-mine-only.feature:12
    And I submit the question                                                    # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-mine-only.feature:13
    And I visit the "explore" page                                               # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-mine-only.feature:14
    And I clear all filters                                                      # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-mine-only.feature:15
    And I clear the search text                                                  # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-mine-only.feature:16
    When I filter documents by mine only = "true"                                # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-mine-only.feature:17
    Then the docs count is less than "100"                                       # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-mine-only.feature:18

@ask-question
Feature: Ask Question

  Any Contributor may ask questions with Markdown formatting, associating one or
  more (existing) "tags" with the question.


  Scenario: Ask a question with Markdown preview      # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:7
    Given I am "Mary"                                 # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:8
    And my user name is "MaryAdmin"                   # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:9
    When I visit the "ask" page                       # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:10
    And I type "" as the question title               # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:11
    Then the submit button is disabled                # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:12
    When I type "foo" as the question title           # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:13
    And I type "**bar**" as the question content      # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:14
    And I enter "javascr" as a question tag           # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:15
    Then the question title is "foo"                  # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:16
    And the question content is "**bar**"             # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:17
    And the question tags menu appears                # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:18
    And the submit button is not disabled             # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:19
    And the previewed content is not displayed        # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:20
    When I click a question tag in the menu           # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:21
    Then the question tag is "javascript"             # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:22
    When I preview the content                        # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:23
    Then the previewed content is displayed           # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:24
    And the previewed content has "strong" formatting # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:25
    And I submit the question                         # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:26
    Then the page title is "doc - samplestack"        # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:27

@vote-on-question
Feature: Vote on Question

  Contributors may vote questions up or down. One may only vote on a question
  once. When an upvote is made to a quesiton, the author of the question
  gains reputation points. When a downvote is made to a  question, the author
  of the question loses reputation points.

@list-docs
Feature: List Docs

  When no search criteria are specified, the explore page lists documents
  **only** according to any applied filters. Such a list is not considered
  a true "search". When listing documents, the results are displayed
  differently. In particular, snippets are not displayed and the default
  sort is different.


  Scenario: As a visitor I can see **some, but not all** of the docs   # :\Users\builder\marklogic-samplestack\specs\features\as-visitor\list-docs.feature:10
    Given I am a visitor                                               # :\Users\builder\marklogic-samplestack\specs\features\as-visitor\list-docs.feature:11
    And I am using the brief seed data                                 # :\Users\builder\marklogic-samplestack\specs\features\as-visitor\list-docs.feature:12
    When I visit the "explore" page                                    # :\Users\builder\marklogic-samplestack\specs\features\as-visitor\list-docs.feature:13
    Then the docs count is "1903"                                      # :\Users\builder\marklogic-samplestack\specs\features\as-visitor\list-docs.feature:14

@explore-docs-by-text @explore
Feature: Explore Docs By Text

  When searching for documents, a text filter may be applied to limit results to
  those documents which match the search text criteria.


  Scenario: As a visitor searching for "Math", I see the correct results                            # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:7
    Given I am a visitor                                                                            # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:8
    And I am using the brief seed data                                                              # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:9
    When I visit the "explore" page                                                                 # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:10
    And I perform a search for "Math"                                                               # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:11
    Then the docs count is "54".                                                                    # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:12
    When I focus on the "first" search result,                                                      # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:13
    Then the result "title" is "Q: How can I detect the distance that the user's mouse has moved?"  # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:14
    When I focus on the "last" search result,                                                       # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:15
    Then the result "title" is "Q: Nested For loop for initializing array of arrays for Javascript" # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:16


  Scenario: As a contributor searching for "Math", I see the correct results                       # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:18
    Given I am a contributor                                                                       # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:19
    And I am using the brief seed data                                                             # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:20
    When I visit the "explore" page                                                                # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:21
    And I perform a search for "Math"                                                              # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:22
    Then the docs count is "77".                                                                   # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:23
    When I focus on the "first" search result,                                                     # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:24
    Then the result "title" is "Q: How can I detect the distance that the user's mouse has moved?" # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:25
    When I focus on the "last" search result,                                                      # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:26
    Then the result "title" is "Q: Length of Number in JavaScript"                                 # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:27

@login
Feature: Login

  The application allows both authenticated and unauthenticated access through
  login and logout operations. Authenticated users have access to more content
  and features.


  Scenario: Try to log in, fail, then succeed        # :\Users\builder\marklogic-samplestack\specs\features\login-logout\login.feature:8
    Given I am a visitor                             # :\Users\builder\marklogic-samplestack\specs\features\login-logout\login.feature:9
    And I am using the brief seed data               # :\Users\builder\marklogic-samplestack\specs\features\login-logout\login.feature:10
    When I start to log in                           # :\Users\builder\marklogic-samplestack\specs\features\login-logout\login.feature:11
    And I attempt to log in with invalid credentials # :\Users\builder\marklogic-samplestack\specs\features\login-logout\login.feature:12
    And my login attempt is denied                   # :\Users\builder\marklogic-samplestack\specs\features\login-logout\login.feature:13
    And I attempt to log in as a Contributor         # :\Users\builder\marklogic-samplestack\specs\features\login-logout\login.feature:14
    Then I am logged in                              # :\Users\builder\marklogic-samplestack\specs\features\login-logout\login.feature:15


  Scenario: Log In as Mary           # :\Users\builder\marklogic-samplestack\specs\features\login-logout\login.feature:17
    Given I am "Mary"                # :\Users\builder\marklogic-samplestack\specs\features\login-logout\login.feature:18
    Then my user name is "MaryAdmin" # :\Users\builder\marklogic-samplestack\specs\features\login-logout\login.feature:19

@view-contributor
Feature: View Contibutor

  Any user may view "contributor detail" about themselves or other contributors
  of content to the system. If the user is a "local contributor", a custom
  dialog displays basic information. If the user is a StackOverflow user, then
  a new window is opened to display that user's StackOverflow profile.


  Scenario: View Contributor Mary                                                # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:9
    Given I am "Mary"                                                            # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:10
    And I am using the brief seed data                                           # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:11
    And I visit the "qnadoc" page with id "5dce8909-0972-4289-93cd-f2e8790a17fc" # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:12
    And I focus on the question                                                  # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:13
    And I view the content contributor                                           # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:14
    Then the contributor display name is "MaryAdmin"                             # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:15
    And the contributor votes cast are greater than "4"                          # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:16
    And the contributor reputation is greater than "99"                          # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:17
    And dismiss the dialog                                                       # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:18


  Scenario: View Contributor Joe                                                 # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:20
    Given I am "Joe"                                                             # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:21
    And I am using the brief seed data                                           # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:22
    And I visit the "qnadoc" page with id "778d0b9c-419f-496a-a300-44815d79708d" # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:23
    And I focus on the question                                                  # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:24
    And I view the content contributor                                           # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:25
    Then the contributor display name is "JoeUser"                               # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:26
    And the contributor votes cast are greater than "2"                          # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:27
    And the contributor reputation is greater than "49"                          # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:28
    And dismiss the dialog                                                       # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:29


(::) failed steps (::)

AssertionError: expected 1903 to equal 1181



Failing scenarios:
C:\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range.feature:7 # Scenario: As a contributor filtering by date range, I see the correct results

12 scenarios (1 failed, 11 passed)
105 steps (1 failed, 7 skipped, 97 passed)
SauceLabs results available at http://saucelabs.com/jobs/da663ef2d50147ecb1ae340491b5d7b5
[launcher] 0 instance(s) of WebDriver still running
[launcher] firefox32 #1 failed 1 test(s)
[launcher] overall: 1 failed spec(s)
[launcher] Process exited with error code 1

C:\Users\builder\marklogic-samplestack\browser>gulp e2e --sauce=win7-firefox-32  --tags="~@broken"

This is passing on sauce and failing locally . Refer to my comments :

Encountered similar failure when running locally with broken flag on.

gulp e2e --browser=ff
:

giving 
    And I view the content contributor                                           # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:14
    Then the contributor display name is "MaryAdmin"                             # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:15
    And the contributor votes cast are greater than "4"                          # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:16
    And the contributor reputation is greater than "99"                          # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:17
    And dismiss the dialog                                                       # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:18


  Scenario: View Contributor Joe                                                 # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:20
    Given I am "Joe"                                                             # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:21
    And I am using the brief seed data                                           # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:22
    And I visit the "qnadoc" page with id "778d0b9c-419f-496a-a300-44815d79708d" # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:23
    And I focus on the question                                                  # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:24
    And I view the content contributor                                           # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:25
    Then the contributor display name is "JoeUser"                               # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:26
    And the contributor votes cast are greater than "2"                          # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:27
    And the contributor reputation is greater than "49"                          # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:28
    And dismiss the dialog                                                       # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:29


12 scenarios (12 passed)
105 steps (105 passed)
Shutting down selenium standalone server.
[launcher] 0 instance(s) of WebDriver still running
[launcher] internet explorer #1 passed

C:\Users\builder\marklogic-samplestack\browser>gulp e2e --browser=ff
[15:04:09] Starting 'clean'...
[15:04:09] Starting 'selenium-start'...
[15:04:09] using **local** Selenium server
[15:04:09] Starting 'middle-tier-start'...
[15:04:09] Finished 'middle-tier-start' after 812 µs
[15:04:09] Finished 'clean' after 414 ms
[15:04:09] Starting 'bower-files'...
selenium standalone is up to date.
chromedriver is up to date.
[15:04:09] Finished 'selenium-start' after 573 ms
[15:04:10] Finished 'bower-files' after 457 ms
[15:04:10] Starting 'build'...
[15:04:10] processing files
[15:04:13] saw 217 files
[15:04:13] Finished 'build' after 3.24 s
Starting selenium standalone server...
[launcher] Running 1 instances of WebDriver
Selenium standalone server started at http://172.18.128.218:49612/wd/hub
@landing-page
Feature: Landing Page

  Users who visit the "home page" of the application should see the explore
  state.


  Scenario: The page title is correct              # :\Users\builder\marklogic-samplestack\specs\features\as-anyone\landing-page.feature:7
    When I visit the "landing" page                # :\Users\builder\marklogic-samplestack\specs\features\as-anyone\landing-page.feature:8
    Then the page title is "explore - samplestack" # :\Users\builder\marklogic-samplestack\specs\features\as-anyone\landing-page.feature:9

@explore-docs-by-resolved-only @explore
Feature: Explore Docs By Resolved Only

  When searching for documents, a facet filter may be applied to limit results to
  those documents which are only resolved.


  Scenario: As a contributor filtering by resolved only, I see the correct results              # :\Users\builder\marklogic-samplestack\specs\features\as-contributor\explore-docs-by-resolved-only.feat
ure:7
    Given I am a contributor                                                                    # :\Users\builder\marklogic-samplestack\specs\features\as-contributor\explore-docs-by-resolved-only.feat
ure:8
    And I am using the brief seed data                                                          # :\Users\builder\marklogic-samplestack\specs\features\as-contributor\explore-docs-by-resolved-only.feat
ure:9
    And I visit the "explore" page                                                              # :\Users\builder\marklogic-samplestack\specs\features\as-contributor\explore-docs-by-resolved-only.feat
ure:10
    And I clear all filters                                                                     # :\Users\builder\marklogic-samplestack\specs\features\as-contributor\explore-docs-by-resolved-only.feat
ure:11
    When I filter documents by resolved only = "true"                                           # :\Users\builder\marklogic-samplestack\specs\features\as-contributor\explore-docs-by-resolved-only.feat
ure:12
    Then the docs count is "1903"                                                               # :\Users\builder\marklogic-samplestack\specs\features\as-contributor\explore-docs-by-resolved-only.feat
ure:13
    When I focus on the "first" search result                                                   # :\Users\builder\marklogic-samplestack\specs\features\as-contributor\explore-docs-by-resolved-only.feat
ure:14
    Then the result "title" is "Q: Mary's Question Number 0"                                    # :\Users\builder\marklogic-samplestack\specs\features\as-contributor\explore-docs-by-resolved-only.feat
ure:15
    When I focus on the "last" search result                                                    # :\Users\builder\marklogic-samplestack\specs\features\as-contributor\explore-docs-by-resolved-only.feat
ure:16
    Then the result "title" is "Q: Show modal if page refreshed, but not if a link is followed" # :\Users\builder\marklogic-samplestack\specs\features\as-contributor\explore-docs-by-resolved-only.feat
ure:17

@explore-docs-by-date-range @explore
Feature: Explore Docs By Date Range

  When searching for documents, a facet filter may be applied to limit results to
  those documents which created by the user.


  Scenario: As a contributor filtering by date range, I see the correct results                                 # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:7
    Given I am "Joe"                                                                                            # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:8
    When I visit the "explore" page                                                                             # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:9
    And I clear all filters                                                                                     # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:10
    When I filter documents by to date = "12/01/2012"                                                           # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:11
    And I press key enter in to date                                                                            # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:12
    Then the docs count is "1181"                                                                               # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:13
      AssertionError: expected 1903 to equal 1181


    When I focus on the "first" search result,                                                                  # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:14
    Then the result "title" is "Q: JS split() function that ignores separator appearing inside quotation marks" # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:15
    When I filter documents by from date = "01/01/2011"                                                         # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:16
    And I press key enter in from date                                                                          # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:17
    Then the docs count is "1004"                                                                               # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:18
    When I focus on the "last" search result,                                                                   # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:19
    Then the result "title" is "Q: Javascript: Easiest way to make a list of elements"                          # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-date-range
.feature:20

@explore-docs-by-mine-only @explore
Feature: Explore Docs By Mine Only

  When searching for documents, a facet filter may be applied to limit results to
  those documents which created by the user.


  Scenario: As a contributor filtering by mine only, I see the correct results   # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-mine-only.feature:7
    Given I am "Joe"                                                             # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-mine-only.feature:8
    When I visit the "ask" page                                                  # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-mine-only.feature:9
    And I type "mine only test" as the question title                            # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-mine-only.feature:10
    And I type "**test**" as the question content                                # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-mine-only.feature:11
    And I enter "e2eTests" as a question tag                                     # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-mine-only.feature:12
    And I submit the question                                                    # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-mine-only.feature:13
    And I visit the "explore" page                                               # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-mine-only.feature:14
    And I clear all filters                                                      # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-mine-only.feature:15
    And I clear the search text                                                  # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-mine-only.feature:16
    When I filter documents by mine only = "true"                                # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-mine-only.feature:17
    Then the docs count is less than "100"                                       # :\Users\builder\marklogic-samplestack\specs\features\as-joe\explore-docs-by-mine-only.feature:18

@ask-question
Feature: Ask Question

  Any Contributor may ask questions with Markdown formatting, associating one or
  more (existing) "tags" with the question.


  Scenario: Ask a question with Markdown preview      # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:7
    Given I am "Mary"                                 # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:8
    And my user name is "MaryAdmin"                   # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:9
    When I visit the "ask" page                       # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:10
    And I type "" as the question title               # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:11
    Then the submit button is disabled                # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:12
    When I type "foo" as the question title           # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:13
    And I type "**bar**" as the question content      # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:14
    And I enter "javascr" as a question tag           # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:15
    Then the question title is "foo"                  # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:16
    And the question content is "**bar**"             # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:17
    And the question tags menu appears                # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:18
    And the submit button is not disabled             # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:19
    And the previewed content is not displayed        # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:20
    When I click a question tag in the menu           # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:21
    Then the question tag is "javascript"             # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:22
    When I preview the content                        # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:23
    Then the previewed content is displayed           # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:24
    And the previewed content has "strong" formatting # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:25
    And I submit the question                         # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:26
    Then the page title is "doc - samplestack"        # :\Users\builder\marklogic-samplestack\specs\features\as-mary\ask-question.feature:27

@vote-on-question
Feature: Vote on Question

  Contributors may vote questions up or down. One may only vote on a question
  once. When an upvote is made to a quesiton, the author of the question
  gains reputation points. When a downvote is made to a  question, the author
  of the question loses reputation points.

@list-docs
Feature: List Docs

  When no search criteria are specified, the explore page lists documents
  **only** according to any applied filters. Such a list is not considered
  a true "search". When listing documents, the results are displayed
  differently. In particular, snippets are not displayed and the default
  sort is different.


  Scenario: As a visitor I can see **some, but not all** of the docs   # :\Users\builder\marklogic-samplestack\specs\features\as-visitor\list-docs.feature:10
    Given I am a visitor                                               # :\Users\builder\marklogic-samplestack\specs\features\as-visitor\list-docs.feature:11
    And I am using the brief seed data                                 # :\Users\builder\marklogic-samplestack\specs\features\as-visitor\list-docs.feature:12
    When I visit the "explore" page                                    # :\Users\builder\marklogic-samplestack\specs\features\as-visitor\list-docs.feature:13
    Then the docs count is "1903"                                      # :\Users\builder\marklogic-samplestack\specs\features\as-visitor\list-docs.feature:14

@explore-docs-by-text @explore
Feature: Explore Docs By Text

  When searching for documents, a text filter may be applied to limit results to
  those documents which match the search text criteria.


  Scenario: As a visitor searching for "Math", I see the correct results                            # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:7
    Given I am a visitor                                                                            # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:8
    And I am using the brief seed data                                                              # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:9
    When I visit the "explore" page                                                                 # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:10
    And I perform a search for "Math"                                                               # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:11
    Then the docs count is "54".                                                                    # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:12
    When I focus on the "first" search result,                                                      # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:13
    Then the result "title" is "Q: How can I detect the distance that the user's mouse has moved?"  # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:14
    When I focus on the "last" search result,                                                       # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:15
    Then the result "title" is "Q: Nested For loop for initializing array of arrays for Javascript" # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:16


  Scenario: As a contributor searching for "Math", I see the correct results                       # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:18
    Given I am a contributor                                                                       # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:19
    And I am using the brief seed data                                                             # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:20
    When I visit the "explore" page                                                                # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:21
    And I perform a search for "Math"                                                              # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:22
    Then the docs count is "77".                                                                   # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:23
    When I focus on the "first" search result,                                                     # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:24
    Then the result "title" is "Q: How can I detect the distance that the user's mouse has moved?" # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:25
    When I focus on the "last" search result,                                                      # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:26
    Then the result "title" is "Q: Length of Number in JavaScript"                                 # :\Users\builder\marklogic-samplestack\specs\features\explore-docs-by-text.feature:27

@login
Feature: Login

  The application allows both authenticated and unauthenticated access through
  login and logout operations. Authenticated users have access to more content
  and features.


  Scenario: Try to log in, fail, then succeed        # :\Users\builder\marklogic-samplestack\specs\features\login-logout\login.feature:8
    Given I am a visitor                             # :\Users\builder\marklogic-samplestack\specs\features\login-logout\login.feature:9
    And I am using the brief seed data               # :\Users\builder\marklogic-samplestack\specs\features\login-logout\login.feature:10
    When I start to log in                           # :\Users\builder\marklogic-samplestack\specs\features\login-logout\login.feature:11
    And I attempt to log in with invalid credentials # :\Users\builder\marklogic-samplestack\specs\features\login-logout\login.feature:12
    And my login attempt is denied                   # :\Users\builder\marklogic-samplestack\specs\features\login-logout\login.feature:13
    And I attempt to log in as a Contributor         # :\Users\builder\marklogic-samplestack\specs\features\login-logout\login.feature:14
    Then I am logged in                              # :\Users\builder\marklogic-samplestack\specs\features\login-logout\login.feature:15


  Scenario: Log In as Mary           # :\Users\builder\marklogic-samplestack\specs\features\login-logout\login.feature:17
    Given I am "Mary"                # :\Users\builder\marklogic-samplestack\specs\features\login-logout\login.feature:18
    Then my user name is "MaryAdmin" # :\Users\builder\marklogic-samplestack\specs\features\login-logout\login.feature:19

@view-contributor
Feature: View Contibutor

  Any user may view "contributor detail" about themselves or other contributors
  of content to the system. If the user is a "local contributor", a custom
  dialog displays basic information. If the user is a StackOverflow user, then
  a new window is opened to display that user's StackOverflow profile.


  Scenario: View Contributor Mary                                                # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:9
    Given I am "Mary"                                                            # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:10
    And I am using the brief seed data                                           # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:11
    And I visit the "qnadoc" page with id "5dce8909-0972-4289-93cd-f2e8790a17fc" # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:12
    And I focus on the question                                                  # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:13
    And I view the content contributor                                           # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:14
    Then the contributor display name is "MaryAdmin"                             # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:15
    And the contributor votes cast are greater than "4"                          # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:16
    And the contributor reputation is greater than "99"                          # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:17
    And dismiss the dialog                                                       # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:18


  Scenario: View Contributor Joe                                                 # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:20
    Given I am "Joe"                                                             # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:21
    And I am using the brief seed data                                           # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:22
    And I visit the "qnadoc" page with id "778d0b9c-419f-496a-a300-44815d79708d" # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:23
    And I focus on the question                                                  # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:24
    And I view the content contributor                                           # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:25
    Then the contributor display name is "JoeUser"                               # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:26
    And the contributor votes cast are greater than "2"                          # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:27
    And the contributor reputation is greater than "49"                          # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:28
    And dismiss the dialog                                                       # :\Users\builder\marklogic-samplestack\specs\features\view-contributor.feature:29


(::) failed steps (::)

AssertionError: expected 1903 to equal 1181



Failing scenarios:

--- a/browser/test/cucumber-support/step-definitions/content.js
+++ b/browser/test/cucumber-support/step-definitions/content.js
@@ -46,9 +46,11 @@ module.exports = function () {
   this.Then(
     /content contributor reputation is greater than "(.*)"/,
     function (repAlias, next) {
+      var self = this;
+
       expect(this.currentPage.focusedItem.metadata.reputation)
-        .to.eventually.be.greaterThan(this[repAlias])
-        .and.notify(next);
+        .to.eventually.be.greaterThan(self[repAlias])
+        .then(this.notifyOk(next), next);
     }
   );
 

--- a/specs/features/as-joe/explore-docs-by-date-range.feature
+++ b/specs/features/as-joe/explore-docs-by-date-range.feature
@@ -4,7 +4,6 @@ Feature: Explore Docs By Date Range
   When searching for documents, a facet filter may be applied to limit results to
   those documents which created by the user.
 
-  @broken @559
   Scenario: As a contributor filtering by date range, I see the correct results
     Given I am "Joe"
     When I visit the "explore" page

--- a/specs/features/as-mary/vote-on-question.feature
+++ b/specs/features/as-mary/vote-on-question.feature
@@ -1,4 +1,4 @@
-@vote-on-question
+@vote-on-question @vote
 Feature: Vote on Question
 
   Contributors may vote questions up or down. One may only vote on a question

--- a/specs/features/explore-docs-by-text.feature
+++ b/specs/features/explore-docs-by-text.feature
@@ -11,7 +11,7 @@ Feature: Explore Docs By Text
     And I perform a search for "Math"
     Then the docs count is "54".
     When I focus on the "first" search result,
-    Then the result "title" is "Q: How can I detect the distance that the user's mouse has moved?"
+    Then the result "title" is "Q: Get element -moz-transform:rotate value in jQuery"
     When I focus on the "last" search result,
     Then the result "title" is "Q: Nested For loop for initializing array of arrays for Javascript"
 
@@ -22,6 +22,6 @@ Feature: Explore Docs By Text
     And I perform a search for "Math"
     Then the docs count is "77".
     When I focus on the "first" search result,
-    Then the result "title" is "Q: How can I detect the distance that the user's mouse has moved?"
+    Then the result "title" is "Q: Get element -moz-transform:rotate value in jQuery"
     When I focus on the "last" search result,
-    Then the result "title" is "Q: Length of Number in JavaScript"
+    Then the result "title" is "Q: Not same src when append images"

